rotkehlchen/globaldb/updates.py

```diff
@@ -402,8 +402,11 @@ def perform_update(
         assert data_directory is not None, 'data directory should be initialized at this point'
         global_db_path = data_directory / 'global_data' / 'global.db'
         with TemporaryDirectory() as tmpdirname:
-            tempdbpath = Path(tmpdirname) / 'temp.db'
-            connection = initialize_globaldb(tempdbpath, GlobalDBHandler().conn.sql_vm_instructions_cb)  # noqa: E501
+            tmpdir = Path(tmpdirname)
+            connection, _ = initialize_globaldb(
+                global_dir=tmpdir,
```

Comment:
why did you change the logic to pass the root dir and not the path?

**Here you introduced a bug that was quite hard to find**.

Thank god for our tests.

You were supposed to initialize a tmpdir/temp.db database with this statement but you do tmpdir/global.db

Reply:
fixed